aws-sdk-s3-transfer-manager/src/operation/upload/service.rs

```diff
@@ -222,6 +222,20 @@ pub(super) async fn read_body(
         .instrument(tracing::debug_span!("read-upload-body"))
         .await?
     {
+        // The content length of each UploadPart request matches the expected part size
+        if part_reader.part_size() != part_data.data.remaining()
+            && part_data.is_last() == Some(false)
```

Comment:
What about a stream that can't tell you if it's the last part? The new method indicates this is a possibility.

Reply:
I suppose we're more lenient in streaming cases in that regard (i.e. the validation won't be triggered). The second guard will become true for cases where `part_data.is_last` can be definitively set to a `Some(...)` variant.; it will always be `None` for streaming cases.

Makes me think whether there is a way to be more strict about streaming cases. Updating the second guard to be `part_data.is_last() == Some(false) || part_data.is_last().is_none()` may trigger validation failure for a streaming case inadvertently where a part is legitimately smaller than configured part size 🤔